dotnet/src/Microsoft.Agents.AI.AzureAI/PersistentAgentsClientExtensions.cs

```diff
@@ -234,4 +375,134 @@ public static ChatClientAgent CreateAIAgent(
+            tools: persistentAgentTools,
+            toolResources: null,
+            temperature: null,
+            topP: null,
+            responseFormat: null,
+            metadata: null,
+            cancellationToken: cancellationToken);
+
+        // Get a local proxy for the agent to work with.
+        return persistentAgentsClient.GetAIAgent(createPersistentAgentResponse.Value.Id, options, clientFactory: clientFactory, cancellationToken: cancellationToken);
+    }
+
+    /// <summary>
+    /// Creates a new server side agent using the provided <see cref="PersistentAgentsClient"/>.
+    /// </summary>
+    /// <param name="persistentAgentsClient">The <see cref="PersistentAgentsClient"/> to create the agent with.</param>
+    /// <param name="model">The model to be used by the agent.</param>
+    /// <param name="options">Full set of options to configure the agent.</param>
+    /// <param name="clientFactory">Provides a way to customize the creation of the underlying <see cref="IChatClient"/> used by the agent.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>A <see cref="ChatClientAgent"/> instance that can be used to perform operations on the newly created agent.</returns>
+    /// <exception cref="ArgumentNullException">Thrown when <paramref name="persistentAgentsClient"/> or <paramref name="model"/> or <paramref name="options"/> is <see langword="null"/>.</exception>
+    /// <exception cref="ArgumentException">Thrown when <paramref name="model"/> is empty or whitespace.</exception>
+    public static async Task<ChatClientAgent> CreateAIAgentAsync(
+        this PersistentAgentsClient persistentAgentsClient,
+        string model,
+        ChatClientAgentOptions options,
+        Func<IChatClient, IChatClient>? clientFactory = null,
+        CancellationToken cancellationToken = default)
+    {
+        if (persistentAgentsClient is null)
+        {
+            throw new ArgumentNullException(nameof(persistentAgentsClient));
+        }
+
+        if (string.IsNullOrWhiteSpace(model))
+        {
+            throw new ArgumentException($"{nameof(model)} should not be null or whitespace.", nameof(model));
+        }
+
+        if (options is null)
+        {
+            throw new ArgumentNullException(nameof(options));
+        }
+
+        List<ToolDefinition> persistentAgentTools = new();
+        if (options.ChatOptions?.Tools is not null)
+        {
+            foreach (AITool tool in options.ChatOptions.Tools)
+            {
+                switch (tool)
+                {
+                    case HostedCodeInterpreterTool:
+                        var codeInterpreterToolDefinition = new CodeInterpreterToolDefinition();
+                        persistentAgentTools.Add(codeInterpreterToolDefinition);
+                        break;
+                }
```

Comment:
We may have other tools to add as a support in here.

See: 
https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/ai/Azure.AI.Agents.Persistent/src/Custom/PersistentAgentsChatClient.cs#L334-L374